src/plugins/content/accessibility/jumplinks/plugin.js

```diff
@@ -50,40 +53,81 @@ export default pluginFactory({
+                    const $element = getJumpElementFactory(areaBroker)[jumpTo];
                     $element.focus();
                 });
                 this.jumplinks.on('shortcuts', () => {
                     this.shortcuts.show();
                     this.shortcuts.getElement()
-                        .off("click", closeShortcutsHandler)
-                        .on("click", closeShortcutsHandler);
+                        .off('click', closeShortcutsHandler)
+                        .on('click', closeShortcutsHandler);
                     $(window)
-                        .off("keydown", closeShortcutsHandler)
-                        .on("keydown", closeShortcutsHandler);
+                        .off('keydown', closeShortcutsHandler)
+                        .on('keydown', closeShortcutsHandler);
                 });
+            })
+            .on('update', function update(params) {
+                this.trigger('changeReviewPanel', params.isReviewPanelEnabled);
+                this.trigger('changeQuesitionStatus', params.questionStatus);
+            })
+            .on('changeReviewPanel', function changeReviewPanel (enabled) {
+                const elem = this.getElement();
+                const panelJumplink = elem
+                    .find('[data-jump="teststatus"]')
+                    .parent();
+                if (enabled) {
+                    panelJumplink.removeClass('hidden');
+                } else {
+                    panelJumplink.addClass('hidden');
+                }
+            })
+            .on('changeQuesitionStatus', function changeQuesitionStatus(questionStatus) {
+                const elem = this.getElement();
+                const text = __('Question') + ' - ' + __(questionStatus);
+                elem
+                    .find('[data-jump="question"] > b')
+                    .text(text);
             });
-        this.shortcuts = shortcutsFactory({});
+        
+        testRunner
+            .on('loaditem', () => {
+                const item = testRunner.getCurrentItem();
+                const config = {
+                    isReviewPanelEnabled: isReviewPanelEnabled(testRunner),
+                    questionStatus: getItemStatus(item)
+                }
+
+                this.jumplinks.trigger('update', config);
+            })
+            .on('tool-flagitem', () => {
+                const item = testRunner.getCurrentItem();
+                const questionStatus = getItemStatus({ ...item, flagged: !item.flagged });
 
+                this.jumplinks.trigger('changeQuesitionStatus', questionStatus);
+            })
+        this.shortcuts = shortcutsFactory({});
     },
 
     /**
      * Called during the runner's render phase
      */
     render: function render() {
-        this.jumplinks.render(this.getAreaBroker().getControlArea());
+        const jumplinksContainer = $('<div class="jump-links-container"></div>');
```

Comment:
Inline template must be avoided. Please use a proper Handlebars template for any HTML to inject.